pandas/core/missing.py

```diff
@@ -219,15 +219,38 @@ def _interpolate_scipy_wrapper(x, y, new_x, method, fill_value=None,
     return new_y
 
 
-def interpolate_2d(values, method='pad', axis=0, limit=None, fill_value=None, dtype=None):
-    """ perform an actual interpolation of values, values will be make 2-d if
-    needed fills inplace, returns the result
+def pad(values, method='pad', axis=0, limit=None, fill_value=None, dtype=None):
+    """ 
+    Perform an actual interpolation of values. 1-d values will be made 2-d temporarily. 
+    Returns the result
     """
 
+    ndim = values.ndim
+    shape = values.shape
+
+    func = partial(pad, method=method, limit=limit, fill_value=fill_value, dtype=dtype)
+
+    if ndim > 2:
+        if ndim == 3:
+            if axis == 0:
+                for n in range(shape[1]):
+                    values[:,n] = func(values[:,n], axis=1)
```

Comment:
this looks quite duplicative. 
this type of if/then is so error prone. just compute the axis to use.